tpot/tpot.py

```diff
@@ -261,16 +267,29 @@ def pareto_eq(ind1, ind2):
 
             self.hof = tools.ParetoFront(similar=pareto_eq)
 
-            stats = tools.Statistics(lambda ind: ind.fitness.values[1])
-            stats.register('Minimum score', np.min)
-            stats.register('Average score', np.mean)
-            stats.register('Maximum score', np.max)
+            # stats = tools.Statistics(lambda ind: ind.fitness.values[1])
+            # stats.register('Minimum score', np.min)
+            # stats.register('Average score', np.mean)
+            # stats.register('Maximum score', np.max)
 
             verbose = (self.verbosity == 2)
 
-            pop, _ = algorithms.eaSimple(population=pop, toolbox=self._toolbox, cxpb=self.crossover_rate,
+            # Start the progress bar
+            num_evaluations = self.population_size * (self.generations + 1)
+            self.pbar = tqdm(total=num_evaluations, unit='pipeline',
+                             disable=(not verbose), desc='GP Progress')
+
+            with warnings.catch_warnings():
+                # Ignore warnings
+                warnings.simplefilter('ignore', category=UserWarning)
+
+                pop, _ = algorithms.eaSimple(population=pop, toolbox=self._toolbox, cxpb=self.crossover_rate,
                                          mutpb=self.mutation_rate, ngen=self.generations,
-                                         stats=stats, halloffame=self.hof, verbose=verbose)
+                                         halloffame=self.hof, verbose=False)
+
+            # Close the progress bar
+            self.pbar.close()
```

Comment:
Good point, it'd be safer to add the call to `pbar.close()` into a `finally` block after the exception handling.